v2/client.go

```diff
@@ -212,3 +214,58 @@ func (c *Client) UserNameLookup(ctx context.Context, usernames []string, opts Us
 		Raw: raw,
 	}, nil
 }
+
+func (c *Client) TweetRecentSearch(ctx context.Context, query string, opts TweetRecentSearchOpts) (*TweetRecentSearchResponse, error) {
+	switch {
+	case len(query) == 0:
+		return nil, fmt.Errorf("tweet recent search: a query is required: %w", ErrParameter)
+	case len(query) > tweetRecentSearchQueryLength:
+		return nil, fmt.Errorf("tweet recent search: the query over the length (%d): %w", tweetRecentSearchQueryLength, ErrParameter)
+	default:
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, tweetRecentSearchEndpoint.url(c.Host), nil)
+	if err != nil {
+		return nil, fmt.Errorf("tweet recent search request: %w", err)
+	}
+	req.Header.Add("Accept", "application/json")
+	c.Authorizer.Add(req)
+	opts.addQuery(req)
+	q := req.URL.Query()
+	q.Add("query", query)
```

Comment:
What is the expected usage of this function?  Is `query` meant to be one search term?  Or can it be a string of url encoded parameters itself?

Reply:
So one of my complaints on the twitter APIs is the documentation.  It had to understand and part of the reason that I have to do a v2.  For example, some of the API documentation do to have all of the objects that may be returned.  So of the examples are not actually output as calling the endpoint through curl results in a different response.
Here is the documentation on the endpoint (https://developer.twitter.com/en/docs/twitter-api/tweets/search/api-reference/get-tweets-search-recent).  It doesn't really say anything about the query, only that it is a search rule and can't be more that 512 characters long.

Also I have started looking at this documentation to help me make sure that I do not make the same mistakes in my API docs.

Reply:
Oh gotcha.  If you think it's possible that the consumer of your sdk would pass in an actual query string, I wonder if it would be better to use `ParseQuery` or something instead to avoid swallowing malformed garbage.
That's a tough call if Twitter isn't clear about expectations.